File: Excel/500-599/504/504 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/504 US Presidents All First Chars Same.xlsx"
input = read_excel(path, range = "A1:A47")
test = read_excel(path, range = "B1:B5")

result = input %>%
  filter(map(`US Presidents`,  ~ length(str_extract_all(., "[A-Z]") %>% 
                                          unlist() %>% unique())) == 1)

identical(result$`US Presidents`, test$`Answer Expected`)
# [1] TRUE

File: Excel/500-599/504/504 Challenge.py
import pandas as pd

path = "504 US Presidents All First Chars Same.xlsx"
input = pd.read_excel(path, usecols="A")
test = pd.read_excel(path, usecols="B", nrows = 4)

result = input[input['US Presidents'].apply(lambda x: len(set([i[0] for i in x.split()])) == 1)].reset_index(drop=True)

print(result["US Presidents"].equals(test["Answer Expected"])) # True